client/ayon_blender/plugins/publish/validate_deadline_publish.py

```diff
@@ -13,6 +14,26 @@
 from ayon_blender.api.render_lib import update_render_product
 
 
+def get_workfile_directory(old_workfile_dir):
+    """Get correct workfile directory for the repair action
+
+    Args:
+        old_workfile_dir (str): old workfile directory
+
+    Returns:
+        str: new_workfile_name
+    """
+    # Use regex to extract the numeric part
+    # Matches one or more digits at the end of the string
+    path = Path(old_workfile_dir)
+
+    old_workfile_name = path.parent.name
+    print(old_workfile_name)
+    old_workfile_match = re.search(r"(\d+)$", old_workfile_name)
+    if old_workfile_match:
+        return old_workfile_name
```

Comment:
I'm still so confused what this is supposed to be doing?

The type hint is incorrect - since it _only_ returns a string if there's a version match. Next question is, why do we even care about the workfile directory and or version in it?

The logic here seems **very** hardcoded to assuming that the render output ALWAYS renders into an output path where the parent folder is the 'workfile name'?

Reply:
I believe the original setup is also hardcoded but thanks for reminding as I find even simpler way for the fix. https://github.com/ynput/ayon-blender/blob/114a09a32fc41d15a560ed58cce61ff00797a6a6/client/ayon_blender/api/render_lib.py#L323